pandas/core/internals/managers.py

```diff
@@ -2274,7 +2275,7 @@ def _preprocess_slice_or_indexer(
 def make_na_array(dtype: DtypeObj, shape: Shape, fill_value) -> ArrayLike:
     if isinstance(dtype, DatetimeTZDtype):
         # NB: exclude e.g. pyarrow[dt64tz] dtypes
-        i8values = np.full(shape, fill_value._value)
+        i8values = np.full(shape, Timestamp(fill_value)._value)
```

Comment:
we need to be sure that the Timestamp here has the same unit as the dtype

Reply:
I added the unit.

Is this function only for NA fill_value? Then it seems unit has no effect since NaT does not have unit.

Reply:
Its always NA for concat, but we can get here with non-NA via reindex with fill_vlaue